src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -1143,6 +1143,36 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupsView {
 		return false;
 	}
 
+	private _globalContextKeys = new Map<RawContextKey<ContextKeyValue>, IContextKey<ContextKeyValue>>();
+	bind<T extends ContextKeyValue>(contextKey: RawContextKey<T>, group: IEditorGroupView): IContextKey<T> {
+		// Ensure we only bind to the same context key once globaly
+		if (!this._globalContextKeys.has(contextKey)) {
+			this._globalContextKeys.set(contextKey, contextKey.bindTo(this.contextKeyService));
+		}
+
+		const scopedContextKeyService = group.scopedContextKeyService;
+		const scopedContextKey = contextKey.bindTo(scopedContextKeyService);
+
+		const $this = this;
+		return {
+			get(): T | undefined {
+				return scopedContextKey.get();
+			},
+			set(value: T): void {
+				if ($this.activeGroup === group) {
+					$this._globalContextKeys.get(contextKey)?.set(value);
```

Comment:
Instead of always looking this up, I would lookup or create the global key in the `bind` method early and then provide access to it.